src/actionmanager.cpp

```diff
@@ -9,9 +9,14 @@
 
 ActionManager::ActionManager(QObject *parent) : QObject(parent)
 {
+    isSaveRecentsEnabled = true;
     recentsListMaxLength = 10;
     openWithMaxLength = 10;
 
+    // Connect to settings signal
+    connect(&qvApp->getSettingsManager(), &SettingsManager::settingsUpdated, this, &ActionManager::settingsUpdated);
```

Comment:
I moved this higher up in the constructor than usual because `loadRecentsList` is called in the middle of the constructor and references some of the settings.